### PR TITLE
contrib/jackc/pgx.v5: add pgx support

### LINKED ISSUE #697: proposal: integrate with github.com/jackc/pgx
How to integrate with https://github.com/jackc/pgx ?

### PR DESCRIPTION
**Pull request by renanferr**
### What does this PR do?

Continuation of #2236

Fixes #697

### Motivation

There wasn't any activity in the mentioned PR for the last weeks so I figured I could fix the tests myself and continue this on another PR.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Comment by samcfinan** [comment]
Also interested to follow this PR's progress. We use pgx and DataDog and would love to use both together.

Thanks @renanferr for moving this along, much appreciated!

**Comment by renanferr** [comment]
Feel free to comment any suggestions, I was working on fixing the tests first.

**Comment by david-azevedo** [comment]
@darccio Thank you for the quick reply! 🚀 

That's awesome! Let us know if there is anything we can help with to move this one along 👍

**Comment by Apologiz** [comment]
> @david-azevedo We are working in our next major release, dd-trace-go v2. We'll discuss internally to include this directly in v2 or in one of the last v1 releases. There is not an exact deadline for v2, but it's going to happen during 2024 Q1.

Do you have any milestone for v2 release?

**Comment by darccio** [comment]
@Apologiz As previously stated, we are working on it but we don't have a fixed release date.

**Comment by rarguelloF** [comment]
@renanferr Thanks for your work fixing the tests, the help is really appreciated!
I went ahead and introduced some changes, adding some missing tags and some other minor refactors.

I've been testing the PR and it LGTM! It can be merged as soon it gets an approval from the folks from @DataDog/apm-go. Thanks everyone for your patience!

**Review by rarguelloF — APPROVED** [review]

**Review by darccio — APPROVED** [review]

File: contrib/jackc/pgx.v5/pgx_tracer.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	_, ctx = tracer.StartSpanFromContext(ctx, "pgx.batch", opts...)
+	return ctx
+}
+
+func (t *pgxTracer) TraceBatchQuery(ctx context.Context, conn *pgx.Conn, data pgx.TraceBatchQueryData) {
+	if !t.cfg.traceBatch {
+		return
+	}
+	// Finish the previous batch query span before starting the next one, since pgx doesn't provide hooks or timestamp
+	// information about when the actual operation started or finished.
```
Comment (darccio) [inline contrib/jackc/pgx.v5/pgx_tracer.go:101]:
```suggestion
	// information about when the actual operation started or finished.
	// pgx.Batch* types don't support concurrency. This function don't support it either.
```

**Comment by darccio** [comment]
This contrib will be released as part of v1.61.0 in the next weeks.

**Comment by oGabrielArruda** [comment]
@darccio Wow! It's great to hear that! 

Why can't you guys realease it at v1.60.2? 😢 
It would be great to use this functionality as soon as possible

**Comment by darccio** [comment]
@oGabrielArruda Patch versions don't include new features. Don't worry, the release process for v1.61.0 starts next Monday 😁

**Comment by samsullivan** [comment]
The timing on this is impeccable; I'm looking forward to next week's release, thanks all!

**Comment by darccio** [comment]
@samsullivan Happy to hear that. Just to keep expectations aligned, it starts next week but it usually takes a few days to complete our QA. We can't give an exact release date, but it'll happen during the next weeks.

Also, updating about v2: we are still working on it and it'll happen during this quarter.

**Pull request merged**